python/ribasim/ribasim/geometry/node.py

```diff
@@ -27,11 +29,20 @@ class NodeSchema(pa.SchemaModel):
 
     class Config:
         add_missing_columns = True
+        coerce = True
 
 
 class Node(SpatialTableModel[NodeSchema]):
     """The Ribasim nodes as Point geometries."""
 
+    # TODO: Remove as soon as add api has been merged
+    @field_validator("df", mode="before")
+    @classmethod
+    def add_node_id_column(cls, df: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        if "node_id" not in df.columns:
+            df.insert(0, "node_id", df.index)
+        return df
```

Comment:
Doesn't this mean it is not breaking? If it is breaking it would be good to have a script to update existing models.

Reply:
Depending on how so see it.
I would say that it's still a breaking change.
Breaking Ribasim Core API is a breaking change, even though Ribasim Python users are spared from the consequences.